### PR TITLE
Sphinx docs

### PR DESCRIPTION
**Pull request by araza6**
Documentation for qgrad

### CODE REVIEW

File: docs/source/index.rst [hunk 1]
```diff
@@ -6,9 +6,25 @@
...
 =================================
 
+**qgrad**
+##########
+
+qgrad is a library that aims to make physics-based optimization tasks easier for the 
+users. qgrad reproduces essential QuTiP functions (with almost the same API) to reduce 
+the friction for existing QuTiP users to transition to a new library. While many
+quantum libraries focus on quantum circuit learning, qgrad focuses on Hamiltonian
+learning type problems. qgrad interfaces with popular machine learning library, JAX
```
Comment (nathanshammah) [inline docs/source/index.rst:16]:
```suggestion
learning type problems. qgrad interfaces with the popular machine learning library, JAX, 
```

File: docs/source/install.rst [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
++----------------+--------------+-----------------------------------------------------+
+| **Matplotlib** | 3.2.2+       | Needed for tutorials.                               |
++----------------+--------------+-----------------------------------------------------+
+| **Cython**     | 0.21+        | Needed for working with QuTiP.                      |  
++----------------+--------------+-----------------------------------------------------+
+| **pytest**     | 5.1.3+       | For running tests                                   |
++----------------+--------------+-----------------------------------------------------+
+
+
+In addition to the standard standard packages listed above,
```
Comment (nathanshammah) [inline docs/source/install.rst:29]:
```suggestion
In addition to the standard packages listed above,
```

File: docs/source/install.rst [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+`JAX <https://github.com/google/jax>`_ is needed to work with qgrad. JAX can be 
+installed with the following commands::
```
Comment (nathanshammah) [inline docs/source/install.rst:31]:
```suggestion
installed with the following commands:
```

File: docs/source/install.rst [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+
```
Comment (nathanshammah) [inline docs/source/install.rst:32]:
```suggestion

.. code-block:: bash

```

File: docs/source/install.rst [hunk 5]
```diff
@@ -0,0 +1,77 @@
...
+    pip install --upgrade pip
+    pip install --upgrade jax jaxlib  # CPU-only version
+
+For custom installing JAX, please visit JAX's GitHub `page <https://github.com/google/jax>`_.
+
+Although qgrad's API is intentionally made similar to that of 
+`QuTiP <https://github.com/qutip/qutip>`_, we have tried our
+best to exclude QuTiP as a dependecy in the core package. We still, however, use
```
Comment (nathanshammah) [inline docs/source/install.rst:40]:
```suggestion
`QuTiP <https://github.com/qutip/qutip>`_, QuTiP is not a dependency of the core package. However, qgrad still requires  
```

Comment (nathanshammah) [reply docs/source/install.rst:40]: Just a suggestion to make the tone more formal.

File: docs/source/install.rst [hunk 6]
```diff
@@ -0,0 +1,77 @@
...
+
+    pip install qutip
+
+For custom installation, see QuTiP `docs <http://qutip.org/docs/4.1/installation.html>`_.
+
+**Installing via pip**
+######################
+
+With the above packages installed, qgrad can now installed via 
+`pip <https://pip.pypa.io/en/stable/>`_::
```
Comment (nathanshammah) [inline docs/source/install.rst:55]: I thought :: was only for python. maybe it works also for bash.

File: docs/source/install.rst (hunk 4 above)
Comment (araza6) [reply docs/source/install.rst:32]: yeah, :: works for bash too. I am wondering then including .. code- block:: bash wouldn't be needed?

**Comment by araza6** [comment]
The `__call__` method in the `Displace` class does not document. Any ideas why?

**Comment by araza6** [comment]
> The `__call__` method in the `Displace` class does not document. Any ideas why?

Even when I manually change the .rst file for Displace, it overwrites it to default upon build.

**Pull request merged**